Updating `osc config` and related topics.

adellape opened the pull request:
Updates to command and file names, re-writes for clarity, and other formatting edits per latest docs guidelines.

@deads2k Can you tech review? Pretty link:

http://file.rdu.redhat.com/~adellape/052015/update_osc_config/cli_reference/manage_cli_profiles.html

deads2k commented:
Looks accurate, but I think we really want to steer people towards `osc login` and `osc project`.  Can you hide the raw `osc config set*` commands in an advanced section or something?

adellape commented:
@deads2k Oh, sure. I take it that leaves `use-context` and `view` as the more commonly useful `osc config` commands to still promote for normal use cases?

deads2k commented:
> Oh, sure. I take it that leaves use-context and view as the more commonly useful osc config commands to still promote for normal use cases?

Let's say only `osc config view`.  `osc project` should be recommended in favor of `osc config use-context`

adellape commented:
@deads2k OK, I was thinking more about the multiple clusters use case for `use-context`, but might as well just use `osc login` in that case. Will rework this topic, thanks kindly.

adellape commented:
@deads2k Updated per feedback, PTAL. Of note:

- Tried different ways of working "Advanced" into the section title for the `osc config` cmds, but ended up renaming it "Manually Configuring CLI Profiles" and added a NOTE admonition to its beginning, explaining its "advanced usage" intent.

- Added a section title "Switching Between CLI Profiles" to house the standard workflow (highlighting `osc login` and `osc project`). That section ends with: "Additional CLI configuration commands are also available for more advanced usage" and a link to the `osc config` cmds section.

- Updated "Example 1. CLI Configuration File" using a latest build.

Pretty link:

http://file.rdu.redhat.com/~adellape/052615/update_osc_config/cli_reference/manage_cli_profiles.html

deads2k commented:
lgtm

adellape commented:
Got peer review from @tnguyen-rh, minor fixes pushed. Will merge after build passes.
